From 1e91faf758d4c27c61c55018a5e6c97b71b107c0 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Thu, 11 Jan 2018 14:53:52 +0200
Subject: [PATCH 1282/1345] crypto: inside-secure: erase used keys

commit  360c95f1fa333767b747cd0ad34de691af67748e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Clear sensative data like used key which could be a security
vulnerability when leaked

Change-Id: Idcd05000f25f98a4af2f692cfab76425a0045282
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48872
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/51663
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/inside-secure/cipher.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/crypto/inside-secure/cipher.c b/drivers/crypto/inside-secure/cipher.c
index 8c3fe48..035f6f8 100644
--- a/drivers/crypto/inside-secure/cipher.c
+++ b/drivers/crypto/inside-secure/cipher.c
@@ -90,6 +90,7 @@ static int safexcel_aes_setkey(struct crypto_ablkcipher *ctfm, const u8 *key,
 
 	ctx->key_len = len;
 
+	memzero_explicit(&aes, sizeof(aes));
 	return 0;
 }
 
@@ -537,10 +538,14 @@ static void safexcel_ablkcipher_cra_exit(struct crypto_tfm *tfm)
 	struct safexcel_crypto_priv *priv = ctx->priv;
 	int ret;
 
+	memzero_explicit(ctx->key, ctx->key_len);
+
 	/* context not allocated, skip invalidation */
 	if (!ctx->base.ctxr)
 		return;
 
+	memzero_explicit(ctx->base.ctxr->data, ctx->key_len);
+
 	/*
 	 * EIP197 has internal cache which needs to be invalidated
 	 * when the context is closed.
-- 
1.7.9.5

